### PR TITLE
IMAGES: Add support for animated tpc mipmaps

### PR DESCRIPTION
**Pull request by Nostritius**
This PR improves the tpc reader for reading the animated tpc file mip maps and not just only the biggest picture

### CODE REVIEW

File: src/images/tpc.cpp [hunk 1]
```diff
@@ -100,6 +100,18 @@ void TPC::readHeader(Common::SeekableReadStream &tpc, byte &encoding) {
 	_layerCount = 2;
 	_isAnimated = checkAnimated(width, height, dataSize);
 
+	if (_isAnimated) {
```
Comment (DrMcCoy) [inline src/images/tpc.cpp]: So animated TPC don't correctly specify the mipmap count in the header, so you need to count them?

Comment (Nostritius) [reply src/images/tpc.cpp]: Yes, exactly. My work here is based on tpc2tga which is doing the exact same thing. Don't ask me what was the intention of creating down to 1x1 mipmaps. But this concept fit perfectly in the remaining size.

File: src/images/tpc.cpp [hunk 2]
```diff
@@ -217,7 +230,7 @@ void TPC::readHeader(Common::SeekableReadStream &tpc, byte &encoding) {
 
 			layerWidth  >>= 1;
 			layerHeight >>= 1;
-			layerSize   >>= 2;
+			layerSize = getDataSize(_format, layerWidth, layerHeight);
```
[the remark this answers is not in the document]
Comment (Nostritius) [reply src/images/tpc.cpp]: Yes, logically speaking, it is almost the same except for the handling of small sizes. which are through that better refined.

**Comment by Nostritius** [comment]
The remaining issue should now be fixed, I also fixed another bug, which i now recognized.

**Comment by DrMcCoy** [comment]
I guess you forgot to remove _offset completely :P

**Comment by Nostritius** [comment]
Damned, yes I forgot one occurence of _offset

**Comment by DrMcCoy** [comment]
Merged as 0af47aa, thanks! :)

Please also have a look at #42, that's unfortunately a regression.

**Pull request closed without merging**